clarity-ext-scripts/clarity_ext_scripts/covid/import_samples.py

```diff
@@ -4,102 +4,127 @@
 from clarity_ext.utils import single
 from clarity_ext.domain import Container, Sample
 
+POSITIVE_PLASMID_CONTROL = "positive plasmid control"
+NEGATIVE_WATER_CONTROL = "negative water control"
+POSITIVE_VIRUS_CONTROL = "positive virus control"
+
+CONTROLS_IN_CSV = {
+    POSITIVE_PLASMID_CONTROL,
+    NEGATIVE_WATER_CONTROL,
+    POSITIVE_VIRUS_CONTROL,
+}
+
 
 class Extension(GeneralExtension):
-    def is_control(self, name):
-        # Returns a control type if the name indicates that this is a control sample. Otherwise
-        # returns None.
-        for control_type, pattern in self.control_mapping.items():
-            if pattern.match(name):
-                return control_type
-        return None
-
-    def create_sample_or_control(self, original_name, project, index):
-        sample_name_fmt = self.context.current_step.udf_sample_name
-        control_name_fmt = self.context.current_step.udf_control_name
-
-        control_type = self.is_control(original_name)
-
-        name_fmt = control_name_fmt if control_type else sample_name_fmt
-        name = format_name(original_name, index, name_fmt, self.context.start)
-        sample_or_control = Sample(sample_id=None, name=name, project=project)
-        sample_or_control.udf_map.force(
-            "Control", "Yes" if control_type else "No")
-        if control_type:
-            sample_or_control.udf_map.force("Control type", control_type)
-        return sample_or_control
+    """
+    Requires two step UDFs:
+        * Assign to workflow: Any workflow 
+        * Project: Any project
 
-    def execute(self):
-        self.control_mapping = dict()
-        for mapping in self.context.current_step.udf_control_mapping.split(","):
-            map_from, map_to = mapping.split(":")
-            self.control_mapping[map_to] = re.compile(map_from)
+    Requires a CSV file with the headers barcode;well 
+
+
+    Creates two containers with samples and controls in Clarity:
+        COVID_PREXT_<timestamp> 
```

Comment:
The naming scripts for the RNA and PCR plates doesn't work together with this naming convention

Reply:
Yeah, but this is what was decided on the meeting yesterday if I'm not mistaken. I guess we'll have to update those then, but I'd prefer to do it in another PR.